src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.Unix.cs

```diff
@@ -62,73 +62,36 @@ internal static unsafe void GetAnsiStringBytes(ReadOnlySpan<char> chars, Span<by
             bytes[actualByteLength] = 0;
         }
 
-        public static IntPtr AllocHGlobal(IntPtr cb)
+        public static unsafe IntPtr AllocHGlobal(IntPtr cb)
         {
-            nuint cbNative = (nuint)(nint)cb;
-
-            // Avoid undefined malloc behavior by always allocating at least one byte
-            IntPtr pNewMem = Interop.Sys.MemAlloc((cbNative != 0) ? cbNative : 1);
-            if (pNewMem == IntPtr.Zero)
-            {
-                throw new OutOfMemoryException();
-            }
-
-            return pNewMem;
+            return (nint)NativeMemory.Alloc((nuint)(nint)cb);
         }
 
-        public static void FreeHGlobal(IntPtr hglobal)
+        public static unsafe void FreeHGlobal(IntPtr hglobal)
         {
-            if (hglobal != IntPtr.Zero)
-            {
-                Interop.Sys.MemFree(hglobal);
-            }
+            NativeMemory.Free((void*)(nint)hglobal);
```

Comment:
This is missing the fast path for null that used to be there before.

It is very convenient to assume that the Free method is very fast for null so that the callers do not have to bother replicating it.

Reply:
I'll add it back to the `NativeMemory` API so it still exists.